packages/sdk-effects/iframe-manager/src/lib/iframe-manager.effects.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { AuthorizeUrl } from '@forgerock/sdk-types';
+
+// Define specific options for the iframe request
+export interface IframeRequestOptions {
+  url: string;
+  timeout: number;
+  // Specify query parameters expected on success/error if needed,
+  // otherwise, parse all parameters found.
+  // Example: expectedParams?: string[];
+}
+
+type ResolvedParams<T> = Record<string, T>;
+
+type Noop = () => void;
+
+/**
+ * Initializes the Iframe Manager effect.
+ * This follows the functional effect pattern, returning an API
+ * within a closure. Configuration could be passed here if needed
+ * for default behaviors (e.g., default timeout), but currently,
+ * all config is per-request.
+ *
+ * @returns An object containing the API for managing iframe requests.
+ */
+export default function iframeManagerInit(/*config: OAuthConfig*/) {
+  /**
+   * Creates an iframe to make a background request to the specified URL,
+   * waits for a redirect, and resolves with the query parameters from the
+   * redirect URL.
+   *
+   * @param options - The options for the iframe request (URL, timeout).
+   * @returns A Promise that resolves with the query parameters from the redirect URL or rejects on timeout or error.
+   */
+  const getAuthCodeByIFrame = (options: {
```

Comment:
Is there a way to make this more generic? In other words, could this just be `getValuesByIframe`? So, this just creates an iframe in a generic way, and detects for the values that are given as arguments. That way, this function doesn't have to know what values to look for or what's relevant. It just manages the iframe, the event listeners, the cleanup, etc. So, it could be useful for getting a Authorization Code or anything else. I'd say, it needs success values and error values to listen for.